djautotask/sync_rest.py

```diff
@@ -563,3 +563,94 @@ def _assign_field_data(self, instance, json_data):
         kwargs['conditions'] = self.api_conditions
         return self.client.get_tasks(next_url, *args, **kwargs)
+
+
+class ProjectSynchronizer(SyncRestRecordUDFMixin, Synchronizer,
+                          ParentSynchronizer):
+    client_class = api.ProjectsAPIClient
+    model_class = models.ProjectTracker
+    udf_class = models.ProjectUDF
+    # override
+    last_updated_field = 'lastActivityDateTime'
+
+    related_meta = {
+        'projectLeadResourceID': (models.Resource, 'project_lead_resource'),
+        'companyID': (models.Account, 'account'),
+        'status': (models.ProjectStatus, 'status'),
+        'projectType': (models.ProjectType, 'type'),
+        'contractID': (models.Contract, 'contract'),
+        'department': (models.Department, 'department'),
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.api_conditions += [
+            ['status', models.ProjectStatus.COMPLETE_ID, 'noteq'],
+            ['status', list(self.get_active_ids()), 'in']
+        ]
+
+    def get_active_ids(self):
+        active_project_statuses = models.ProjectStatus.objects.exclude(
+            is_active=False).values_list('id', flat=True).order_by(
+            self.lookup_key)
+
+        return active_project_statuses
+
+    def _assign_field_data(self, instance, object_data):
+
+        instance.id = object_data['id']
+        instance.name = object_data.get('projectName')
+        instance.number = object_data.get('projectNumber')
+        instance.description = object_data.get('description')
+        instance.actual_hours = object_data.get('actualHours')
+        instance.completed_percentage = object_data.get('completedPercentage')
+        instance.duration = object_data.get('duration')
+        instance.estimated_time = object_data.get('estimatedTime')
+        instance.status_detail = object_data.get('statusDetail')
+
+        instance.completed_date = object_data.get('completedDateTime')
+        instance.end_date = object_data.get('endDateTime')
+        instance.start_date = object_data.get('startDateTime')
+        instance.last_activity_date_time = \
+            object_data.get('lastActivityDateTime')
+
+        self._set_datetime_attribute(instance, 'completed_date')
+        self._set_datetime_attribute(instance, 'end_date')
+        self._set_datetime_attribute(instance, 'start_date')
+        self._set_datetime_attribute(instance, 'last_activity_date_time')
+
+        if instance.completed_date:
+            instance.completed_date = instance.completed_date.date()
+        if instance.end_date:
+            instance.end_date = instance.end_date.date()
+        if instance.start_date:
+            instance.start_date = instance.start_date.date()
+
+        udfs = object_data.get('userDefinedFields')
+
+        # Refresh udf field to eliminate stale udfs
+        instance.udf = dict()
+
+        if len(udfs):
+            self._assign_udf_data(instance, udfs)
```

Comment:
If UDF's are handled here, can you removed `ProjectUDFSynchronizer`?

Reply:
Oh nevermind. I always forget how UDFs get synced. But I think we still need the `UDFSynchronizer`s in sync.py. That should probably have its own issue. I'll add that later.

Reply:
Okay, we can dig into syncing UDFs more in that issue.